packages/ember-metal/lib/property_set.js

```diff
@@ -62,8 +62,11 @@ export function set(obj, keyName, value, tolerant) {
     return setPath(obj, keyName, value, tolerant);
   }
 
-  assert(`calling set on destroyed object: ${toString(obj)}.${keyName} = ${toString(value)}`,
-         !obj.isDestroyed);
+  if (obj.isDestroyed) {
```

Comment:
basically 3 scenarios exist.

1. set on plain object property
2. set on descriptor
3. set on deep object

We currently guard/assert for 1, and I believe we must absolutely also guard/assert for 2. I also believe we should consider the same for 3, if others disagree I would like to be convinced.